server/schema/media/media.resolver.js

```diff
@@ -14,4 +15,35 @@ module.exports = {
       throw APIError(null, error);
     }
   },
+  addMedia: async (
+    _parent,
+    { authors, store, storePath, mediaType, blurhash },
+    { mid, session, authToken, decodedToken, API: { Media } }
+  ) => {
+    try {
+      if (!UserPermission.exists(session, authToken, decodedToken, 'media.write.self')) {
+        throw APIError('FORBIDDEN', null, {
+          reason: 'The user does not have the required permission to perform this operation.',
+        });
+      }
+      const media = Media.create(authors, store, storePath, mediaType, blurhash, session, authToken, mid);
+      return media;
+    } catch (error) {
+      throw APIError(null, error);
+    }
+  },
+  deleteMediaById: async (_parent, { id }, { mid, session, authToken, decodedToken, API: { Media } }) => {
+    try {
+      if (!UserPermission.exists(session, authToken, decodedToken, 'media.write.all')) {
```

Comment:
This should check for both `media.write.all` and `media.write.self` if the image was uploaded by the same user.